backend/src/apiserver/server/run_server.go

```diff
@@ -30,6 +35,41 @@ type RunServer struct {
 
 func (s *RunServer) CreateRun(ctx context.Context, request *api.CreateRunRequest) (*api.RunDetail, error) {
 	err := s.validateCreateRunRequest(request)
+	if err != nil {
```

Comment:
Most of the logic here are common functionalities. Could you extract the auth logic to 
https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/server/util.go

Reply:
I extracted some of the common logic to the util. The whole if-block is createrun specific in the same sense of extracting the namespace from api.RunDetail.

Reply:
Please extract the common logic and make sure they are covered by tests